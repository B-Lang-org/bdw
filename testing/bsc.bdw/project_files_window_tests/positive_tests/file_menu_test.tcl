
# Creates a new project
handlers::open_project project.bspec 

# Sets the compilation result directories
handlers::set_compilation_results_location -simdir mult4 -bdir mult4

# Displays the project status
commands::display_project_status

# Cleans the files
commands::full_clean

while {$BSPEC(FILES) == ""} {
        update
}

# Compiles the specified file considering dependencies 
handlers::compile_file mult4/Mult2.bsv -withdeps -typecheck

while {$BSPEC(BUILDPID) != ""} {
        update
}

# Cleans the files
commands::full_clean

while {$BSPEC(FILES) == ""} {
        update
}

# Compiles the specified file considering dependencies 
handlers::compile_file mult4/Mult2.bsv -withdeps

while {$BSPEC(BUILDPID) != ""} {
        update
}

# Compiles the specified file without considering dependencies 
handlers::compile_file mult4/Mult2.bsv 

while {$BSPEC(BUILDPID) != ""} {
        update
}

# Cleans the files
commands::full_clean

# Closes the project 
handlers::close_project

# Exits the worksptation
exit
