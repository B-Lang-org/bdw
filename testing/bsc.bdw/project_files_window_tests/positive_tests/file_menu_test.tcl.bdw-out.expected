COMP_BDIR = mult4
COMP_BSC_OPTIONS = -keep-fires
COMP_BSC_TYPE = bluesim
COMP_INFO_DIR = .
COMP_SIMDIR = mult4
COMP_TYPE = bsc
COPY_FLAGS = 0
COMP_VDIR = .
DIR = .
EDITOR_NAME = gvim
INCLUDED_FILES = *.bsv
LINK_BSC_OPTIONS = -keep-fires
LINK_OUTDIR = .
LINK_OUTNAME = out
LINK_TYPE = bsc
MAIN_PLACEMENT = 700x350+0+0
MODULE_PLACEMENT = 550x350+50+500
NAME = project
PATHS = mult4
PATHS = .
PATHS = Libraries
PACKAGE_PLACEMENT = 770x420+475+150
PROJECT_PLACEMENT = 250x460+0+385
SCHEDULE_PLACEMENT = 650x650+500+225
SIM_NAME = iverilog
TOP_FILE = mult4/Mult4Tb.bsv
TOP_MODULE = mkMult4Tb
TYPE_PLACEMENT = 770x420+475+540
VIEWER_CLOSE = close
WIZARD_PLACEMENT = 1024x700+50+50
EDITOR_GVIM = gvim
EDITOR_EMACS = emacs
EDITOR_OTHER = gedit +%n %f
Executing: bsc -u  -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries  mult4/Mult2.bsv
WARNING: + bsc -u -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries mult4/Mult2.bsv
RESULT: checking package dependencies
RESULT: compiling mult4/Mult.bsv
RESULT: compiling mult4/Mult2.bsv
RESULT: All packages are up to date.
Executing: bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries  mult4/Mult2.bsv
WARNING: + bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries mult4/Mult2.bsv
RESULT: checking package dependencies
RESULT: compiling mult4/Mult.bsv
RESULT: compiling mult4/Mult2.bsv
RESULT: code generation for mkMult2 starts
WARNING: Warning: "mult4/Mult2.bsv", line 23, column 8: (G0117)
WARNING:   Rule `start' shadows the effects of `cycle' when they execute in the same
WARNING:   clock cycle. Affected method calls:
WARNING:     product.write, mcand.write, mplr.write
WARNING:   To silence this warning, use the `-no-warn-action-shadowing' flag.
RESULT: Elaborated module file created: mult4/mkMult2.ba
RESULT: All packages are up to date.
Executing: bsc -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries  mult4/Mult2.bsv
WARNING: + bsc -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries mult4/Mult2.bsv
WARNING: Warning: "mult4/Mult2.bsv", line 23, column 8: (G0117)
WARNING:   Rule `start' shadows the effects of `cycle' when they execute in the same
WARNING:   clock cycle. Affected method calls:
WARNING:     product.write, mcand.write, mplr.write
WARNING:   To silence this warning, use the `-no-warn-action-shadowing' flag.
RESULT: Elaborated module file created: mult4/mkMult2.ba
